velox/exec/tests/AggregationTest.cpp

```diff
@@ -1591,6 +1592,32 @@ TEST_F(AggregationTest, groupingSetsOutput) {
   assertEqualResults(orderResult.second, reversedOrderResult.second);
 }
 
+TEST_F(AggregationTest, groupingSetsSameKey) {
+  auto data = makeRowVector(
+      {"o_key", "o_status"},
+      {makeFlatVector<int64_t>({0, 1, 2, 3, 4}),
+       makeFlatVector<std::string>({"", "x", "xx", "xxx", "xxxx"})});
+
+  createDuckDbTable({data});
+
+  auto plan = PlanBuilder()
+                  .values({data})
+                  .groupId(
+                      {"o_key", "o_key as o_key_1"},
```

Comment:
I still find this use case confusing. It is just not clear why would anything use a plan like this. I wish we could identify a compelling use case.

Reply:
I felt the query documented with its Presto plan was a better example. But I wasn't able to translate the IF expressions to equivalent CASE expressions. `case group_id when 1 then 0 else null end` could not compile since null is UNKNOWN type and CASE required all when/then expressions to evaluate to the same type.

Reply:
try "null::bigint"

Reply:
I will send out a follow up PR.